models/room.js

```diff
@@ -217,10 +217,14 @@ RoomSchema.methods.look = function (socket, short) {
     output += `<span class="silver">${this.desc}</span>\n`;
   }
 
-  if (this.inventory && this.inventory.length > 0) {
+  if (this.inventory && this.inventory.filter(item => !item.hidden).length > 0) {
     output += `<span class="darkcyan">You notice: ${this.inventory.filter(item => !item.hidden).map(item => item.displayName).join(', ')}.</span>\n`;
```

Comment:
You don't want to filter the the inventory twice if you don't have to, if you end up taking the filter out of the if statement above this line, then this comment is moot.